### PR TITLE
update parser tests, fix small bugs and corner cases discovered

### PR DESCRIPTION
**Pull request by riftgarret**
* added tests for all ParserHelper methods.
* added corner case exceptions for methods that should be checked
* updated creating the TargetRule
* added json library to unit test

### CODE REVIEW

File: Redninja/Components/Targeting/TargetPatternFactory.cs [hunk 1]
```diff
@@ -36,6 +36,24 @@ public bool ContainsLocation(Coordinate anchor, int targetRow, int targetColumn)
 
 			public bool ContainsLocation(Coordinate anchor, Coordinate target)
 				=> ContainsLocation(anchor.Row, anchor.Column, target.Row, target.Column);
+
+			public override bool Equals(object obj)
```
Comment (riftgarret) [inline Redninja/Components/Targeting/TargetPatternFactory.cs:40]: i dont know if this is ideal to stick this directly into this class. It serves the purpose to compare against in the UnitTests but like you said, its unrealistic to be a true comparable. For now we are just checking from anchor point.

**Review by RiceGnat — APPROVED** [review]
